lib/src/main/kotlin/at/bitfire/ical4android/Event.kt

```diff
@@ -127,17 +132,26 @@ class Event: ICalendar() {
 
             val events = mutableListOf<Event>()
             for ((uid, vEvent) in mainEvents) {
-                val event = fromVEvent(vEvent)
+                try {
+                    val event = fromVEvent(vEvent)
 
-                // assign exceptions to main event and then remove them from exceptions array
-                exceptions.remove(uid)?.let { eventExceptions ->
-                    event.exceptions.addAll(eventExceptions.values.map { fromVEvent(it) })
-                }
+                    // assign exceptions to main event and then remove them from exceptions array
+                    exceptions.remove(uid)?.let { eventExceptions ->
+                        event.exceptions.addAll(eventExceptions.values.map { fromVEvent(it) })
+                    }
+
+                    // make sure that exceptions have at least a SUMMARY
+                    event.exceptions.forEach { it.summary = it.summary ?: event.summary }
 
-                // make sure that exceptions have at least a SUMMARY
-                event.exceptions.forEach { it.summary = it.summary ?: event.summary }
+                    events += event
+                } catch (e: InvalidCalendarException) {
+                    Ical4Android.log.warning("Invalid VEvent: e")
+                    Ical4Android.log.warning(vEvent.toString())
```

Comment:
You can do `Ical4Android.log.log(Level.WARNING, "Invalid event: $vEvent", e)`, then we have the exception in the same line

Reply:
That's way better, fixed